File: ios/Classes/enums/PendencyTypeEnum.swift
import ImSDK;

//  Created by 蒋具宏 on 2020/2/14.
//  未决类型枚举
public enum PendencyTypeEnum : Int{
    /**
     *  别人发给我的未决请求
     */
    case COME_IN = 1
    
    /**
     *  我发给别人的未决请求
     */
    case SEND_OUT = 2
    
    /**
     *  别人发给我的以及我发给别人的所有未决请求，仅在拉取时有效。
     */
    case BOTH = 3
    
    /**
     * 根据腾讯云获得枚举
     */
    static func getByTIMPendencyType(type : TIMPendencyType)->PendencyTypeEnum{
        return (PendencyTypeEnum(rawValue: type.rawValue) ?? nil)!;
    }
}

File: lib/enums/pendency_type_enum.dart
/// 未决类型
enum PendencyTypeEnum {
  /// 别人发给我的未决请求
  COME_IN,

  /// 我发给别人的未决请求
  SEND_OUT,

  /// 别人发给我的以及我发给别人的所有未决请求，仅在拉取时有效。
  BOTH,
}

/// 枚举工具
class PendencyTypeTool {
  /// 根据数标获得枚举
  static PendencyTypeEnum getEnumByIndex(index) {
    switch (index) {
      case 1:
        return PendencyTypeEnum.COME_IN;
      case 2:
        return PendencyTypeEnum.SEND_OUT;
      case 3:
        return PendencyTypeEnum.BOTH;
      default:
        return null;
    }
  }

  /// 根据枚举获得数标
  static int getIndexByEnum(e) {
    switch (e) {
      case PendencyTypeEnum.COME_IN:
        return 1;
      case PendencyTypeEnum.SEND_OUT:
        return 2;
      case PendencyTypeEnum.BOTH:
        return 3;
      default:
        return null;
    }
  }
}
